jerry-core/jerry.c

```diff
@@ -914,6 +914,56 @@ jerry_create_object (void)
 } /* jerry_create_object */
 
 /**
+ * Create string from a valid UTF8 string
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value when it is no longer needed.
+ *
+ * @return value of the created string
+ */
+jerry_value_t
+jerry_create_string_from_utf8 (const jerry_char_t *str_p) /**< pointer to string */
+{
+  return jerry_create_string_sz_from_utf8 (str_p, lit_zt_utf8_string_size ((lit_utf8_byte_t *) str_p));
+} /* jerry_create_string_from_utf8 */
+
+/**
+ * Create string from a valid UTF8 string
+ *
+ * Note:
+ *      returned value must be freed with jerry_release_value when it is no longer needed.
+ *
+ * @return value of the created string
+ */
+jerry_value_t
+jerry_create_string_sz_from_utf8 (const jerry_char_t *str_p, /**< pointer to string */
+                                  jerry_size_t str_size) /**< string size */
+{
+  jerry_assert_api_available ();
+
+  ecma_string_t *ecma_str_p = NULL;
+  lit_utf8_size_t dest_buf_size = 0;
+
+  dest_buf_size = lit_calculate_cesu8_size_from_utf8 ((lit_utf8_byte_t *) str_p, (lit_utf8_size_t) str_size);
+
+  if ((lit_utf8_size_t) str_size != dest_buf_size)
+  {
+    lit_utf8_byte_t dest_buf[dest_buf_size];
```

Comment:
Indeed, but I think we should use JMEM_DEFINE_LOCAL_ARRAY also and it wouldn't require too much modification, what is your opinion?